### ISSUE TITLE
TypeScript suggestions skip code without semicolon

### ISSUE THREAD

**Issue by svipas** [microsoft/vscode#184847]
Type: <b>Bug</b>

Latest VS Code and TS doesn't show completion for code if it doesn't have semicolon.

<img width="259" alt="image" src="https://github.com/microsoft/vscode/assets/10441177/2f129a62-167a-47a6-aebd-8a875e7693fe">

<img width="225" alt="image" src="https://github.com/microsoft/vscode/assets/10441177/eec47b15-44fe-40f3-a5c6-42af21bea57b">

As an example take this code snippet:

```ts
const userName = "Benas" // if I add ; at the end here it will work

user| // | - is the cursor
```

You will see that it doesn't show `userName` in the suggestions, it will show it only after I add `;` at the end of variable creation. This started to happen with latest VS Code which ships with latest TS (5.1.3). In previous TS version it was working fine.

VS Code version: Code 1.79.0 (b380da4ef1ee00e224a15c1d4d9793e27c2b6302, 2023-06-07T14:31:03.317Z)
OS version: Darwin arm64 22.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|32.00GB (7.57GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.4.0
gitlens|eam|13.6.0
prettier-vscode|esb|9.13.0
copilot|Git|1.88.132
vscode-graphql|Gra|0.8.13
vscode-graphql-syntax|Gra|1.2.0
symbols|mig|0.0.11
vscode-react-native|msj|1.11.0
vscode-thunder-client|ran|2.7.6
vscode-fileutils|sle|3.10.1
vim|vsc|1.25.2
quokka-vscode|Wal|1.0.543

(1 theme extensions excluded)

</details>

**Comment by mayhsundar** [comment]
Hi @jrieken 

It seems like the reason behind this issue might be that VS Code is relying on semicolons to identify the end of statements in TypeScript. When a statement doesn't have a semicolon, VS Code can't accurately determine where it ends and therefore won't be able to suggest any relevant variables for that statement.

In contrast to TypeScript, Python doesn't require semicolons to terminate statements although some people use them for clarity. If you're working with VS Code and want it to suggest variables, you'll need to include semicolons in your TypeScript code.

Please feel free to correct me. :)

**Comment by svipas** [comment]
@mayhsundar JS has ASI so semicolons are not required and as mentioned this was working previously.

**Comment by mayhsundar** [comment]
@svipas okay thanks for the clarity. 👍

**Comment by jhaemin** [comment]
Duplicate of https://github.com/microsoft/vscode/issues/184786

**Comment by IllusionMH** [comment]
/duplicate of #184786

Thanks @jhaemin

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! We figured it's covering the same as another one we already have. Thus, we closed this one as a duplicate. You can search for [similar existing issues](https://github.com/microsoft/vscode/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+). See also our [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Happy Coding!
